src/dmd/semantic3.d

```diff
@@ -1289,15 +1289,14 @@ private extern(C++) final class Semantic3Visitor : Visitor
             assert(f.parameterList.length == funcdecl.parameters.dim);
             foreach (u, v; *funcdecl.parameters)
             {
-                if (v.storage_class & STC.maybescope)
-                {
-                    //printf("Inferring scope for %s\n", v.toChars());
-                    Parameter p = f.parameterList[u];
-                    notMaybeScope(v);
-                    v.storage_class |= STC.scope_ | STC.scopeinferred;
-                    p.storageClass |= STC.scope_ | STC.scopeinferred;
-                    assert(!(p.storageClass & STC.maybescope));
-                }
+                if (!(v.storage_class & STC.maybescope))
```

Comment:
I'm not too thrilled about changing all the code base over things like this. For one thing, it adds a ! which is harder to cognitively process. Having straightforward positive expressions is better.